internals/secrethub/debug.go

```diff
@@ -1,54 +1,17 @@
 package secrethub
 
 import (
-	"strconv"
-
 	"github.com/secrethub/secrethub-cli/internals/cli"
+	"github.com/spf13/cobra"
 )
 
 // RegisterDebugFlag registers a debug flag that changes the log level of the given logger to DEBUG.
 func RegisterDebugFlag(app *cli.App, logger cli.Logger) {
-	flag := debugFlag{
-		logger: logger,
-	}
-	app.PersistentFlags().VarP(&flag, "debug", "D", "Enable debug mode.")
-	app.Root.Flag("debug").NoOptDefVal = "true"
-}
-
-// debugFlag configures the debug level of a logger.
-type debugFlag struct {
-	debug  bool
-	logger cli.Logger
-}
-
-func (f debugFlag) Type() string {
-	return "debugFlag"
-}
-
-func (f debugFlag) init() {
-	if f.debug {
-		f.logger.EnableDebug()
-	}
-}
-
-// String implements the flag.Value interface.
-func (f debugFlag) String() string {
-	return strconv.FormatBool(f.debug)
-}
-
-// Set changes the log level to debug when the given value is true.
-func (f *debugFlag) Set(value string) error {
-	b, err := strconv.ParseBool(value)
-	if err != nil {
-		return err
-	}
-	f.debug = b
-	f.init()
-	return nil
-}
-
-// IsBoolFlag makes the flag a boolean flag when used in a Kingpin application.
-// Thus, the flag can be used without argument (--debug or -D).
-func (f debugFlag) IsBoolFlag() bool {
-	return true
+	app.PersistentFlags().BoolP("debug", "D", false, "Enable debug mode.")
+	app.Root.Cmd.PersistentPreRunE = cli.MergeFuncs(app.Root.Cmd.PersistentPreRunE, func(command *cobra.Command, strings []string) error {
+		if app.Root.Cmd.Flag("debug").Changed {
+			logger.EnableDebug()
```

Comment:
Yes! This is way shorter!

However, we should be very careful with reusing strings like currently done with `"debug"`. If the name of the flag is changed in one place but not the other, we have introduced a bug.


Is there a specific reason for doing it this way? If not, an easy solution would be to do:
```
flag := app.PersistentFlags().BoolP("debug", "D", false, "Enable debug mode.")
[...]
if flag.Changed{
...
}
```

This avoids the requirement of two distinct strings being equal.

Reply:
Good idea to avoid accidental bugs. 
Adjusted the code for both `debug` and `mlock` flags.